experimental/transliteration/src/provider.rs

```diff
@@ -64,9 +64,24 @@ pub struct RuleBasedTransliterator<'a> {
     /// The list of conversion rule groups this transliterator uses.
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub rule_group_list: VarZeroVec<'a, VarZeroSlice<RuleULE>>,
-    /// The direct dependencies of this transliterator.
-    #[cfg_attr(feature = "serde", serde(borrow))]
-    pub dependencies: VarZeroVec<'a, str>,
+}
+
+impl<'a> RuleBasedTransliterator<'a> {
+    /// Returns an iterator of dependencies on other transliterators.
+    ///
+    /// Note that this may contain duplicate entries.
+    pub fn deps(&'a self) -> impl Iterator<Item = SimpleId<'a>> {
```

Comment:
A `SimpleId` is an ID and a filter. I can't think of a scenario where we care what filter is used per dependency?

Suggestion: replace iterator item type with `&'a str`.